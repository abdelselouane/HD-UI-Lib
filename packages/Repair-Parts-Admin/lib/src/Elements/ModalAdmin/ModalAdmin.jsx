import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Modal = styled.div`
  .card .card-toolbar i {
    ${props => props.modalType === 'success' && 'color: #fff !important;' }
    font-size: 20px;
    margin-top: 10px;
  }
`;

const ModalAdmin = (props) => {

  if(!props.show) {
    return null;
  }

  const {
    modal,
    toolBar,
    content,
    actions,
    styles
  } = props;

  const typeColor = modal.type === "success" ? "#4DAA58" : "#DA3832";

  const modalStyles = {
    display: 'block',
    width: styles.width,
    left: '37%',
    top: styles.top ? styles.top : '30%',
    padding: modal.type && 0
  };

  const backdropStyle = {
    position: 'fixed',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: 'rgba(0,0,0,0.3)',
    padding: 50
  };

  const toolBarStyles = {
    marginBottom: 0,
    backgroundColor: modal.type && typeColor,
    padding: modal.type && "10px",
    color: modal.type && "#fff",
  }

  const actionsStyles = {
    padding: "10px",
    margin: 0
  }

  const cardContent = {
    paddingRight: '20px'
  }

  return (
    <Modal className="modal-container" style={backdropStyle} modalType={modal.type}>
      <div className="card modal" style={modalStyles} id={modal.id}>
        <div className="card-toolbar" style={toolBarStyles}>
            <span className="card-title">
                <h3>{
                    modal.type &&
                    <i className={modal.type === "success" ? "icon_check-circle-outlined" : "icon_ban"}></i>
                  }
                    {` ${toolBar.title}`}</h3>
              </span>
            <i className="icon_close" onClick={toolBar.onClick}></i>
          </div>
        <div className="card-content" style={cardContent}>
            { content }
          </div>
        <div className="card-actions right" style={actionsStyles}>
            { actions &&
            actions.map( (action, index) => {
              return (
                <button
                  id={action.id}
                  key={index}
                  className={action.className}
                  onClick={action.onClick}
                  disabled={action.disabled && "disabled"} >
                  {action.text}
                </button>
              )
            })
          }
          </div>
      </div>
    </Modal>
    );
}

ModalAdmin.propTypes = {
  modal: PropTypes.shape({
    id: PropTypes.isRequired
  }),
  toolBar: PropTypes.object,
  actions: PropTypes.arrayOf(PropTypes.object),
  styles: PropTypes.object
}

export default ModalAdmin;