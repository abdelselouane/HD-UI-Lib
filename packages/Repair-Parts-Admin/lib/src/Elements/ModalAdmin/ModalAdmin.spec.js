import React from 'react';
import { shallow } from 'enzyme';
import ModalAdmin, { Modal } from './ModalAdmin';
import toJson from 'enzyme-to-json';
import * as Enzyme from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

describe('<ModalAdmin />', () => {
    let wrapper;
    let props;

    afterEach(() => {
        jest.clearAllMocks();
    });

    beforeAll(() => {
      // ReactDOM.createPortal = jest.fn();
        props = {
          modal: {
            id: "modalID",
            type: ""
          },
          toolBar: {
            title: `Edit cost of part #123`,
            onClick: jest.fn(),
            ref: jest.fn()
          },
          content: 'content',
          actions: [{
            id: 'close',
            onClick: jest.fn(),
            className: "button",
            text: "Cancel"
          },
          {
            id: 'save',
            onClick: jest.fn(),
            className: `button primary`,
            text: "Save Item"
          }],
          styles: {
            width: "500px",
            height: "280px"
          }
        };
        wrapper = shallow(<ModalAdmin {...props} show={true} />);
    });
  
    it('should render properly', () => {
      wrapper = shallow(<ModalAdmin {...props} />);
      wrapper.setProps({ show: false });
      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.setProps({ show: true });
      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.setProps({ modal: { ...props.modal, type: 'success' } });
      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.setProps({ modal: { ...props.modal, type: 'failure' } });
      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.setProps({ styles: { ...props.styles, top: '0%' } });
      expect(toJson(wrapper)).toMatchSnapshot();
      const newActions = props.actions;
      newActions[0] = {
        ...newActions[0],
        disabled: true
      };
      wrapper.setProps({ actions: newActions });
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(toJson(shallow(<Modal modalType={"success"} />))).toMatchSnapshot();
    });

});