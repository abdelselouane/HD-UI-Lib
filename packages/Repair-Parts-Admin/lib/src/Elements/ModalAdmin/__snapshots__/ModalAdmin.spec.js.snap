// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalAdmin /> should render properly 1`] = `""`;

exports[`<ModalAdmin /> should render properly 2`] = `
<styled.div
  className="modal-container"
  modalType=""
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.3)",
      "bottom": 0,
      "left": 0,
      "padding": 50,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="card modal"
    id="modalID"
    style={
      Object {
        "display": "block",
        "left": "37%",
        "padding": "",
        "top": "30%",
        "width": "500px",
      }
    }
  >
    <div
      className="card-toolbar"
      style={
        Object {
          "backgroundColor": "",
          "color": "",
          "marginBottom": 0,
          "padding": "",
        }
      }
    >
      <span
        className="card-title"
      >
        <h3>
           Edit cost of part #123
        </h3>
      </span>
      <i
        className="icon_close"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="card-content"
      style={
        Object {
          "paddingRight": "20px",
        }
      }
    >
      content
    </div>
    <div
      className="card-actions right"
      style={
        Object {
          "margin": 0,
          "padding": "10px",
        }
      }
    >
      <button
        className="button"
        id="close"
        key="0"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button primary"
        id="save"
        key="1"
        onClick={[MockFunction]}
      >
        Save Item
      </button>
    </div>
  </div>
</styled.div>
`;

exports[`<ModalAdmin /> should render properly 3`] = `
<styled.div
  className="modal-container"
  modalType="success"
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.3)",
      "bottom": 0,
      "left": 0,
      "padding": 50,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="card modal"
    id="modalID"
    style={
      Object {
        "display": "block",
        "left": "37%",
        "padding": 0,
        "top": "30%",
        "width": "500px",
      }
    }
  >
    <div
      className="card-toolbar"
      style={
        Object {
          "backgroundColor": "#4DAA58",
          "color": "#fff",
          "marginBottom": 0,
          "padding": "10px",
        }
      }
    >
      <span
        className="card-title"
      >
        <h3>
          <i
            className="icon_check-circle-outlined"
          />
           Edit cost of part #123
        </h3>
      </span>
      <i
        className="icon_close"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="card-content"
      style={
        Object {
          "paddingRight": "20px",
        }
      }
    >
      content
    </div>
    <div
      className="card-actions right"
      style={
        Object {
          "margin": 0,
          "padding": "10px",
        }
      }
    >
      <button
        className="button"
        id="close"
        key="0"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button primary"
        id="save"
        key="1"
        onClick={[MockFunction]}
      >
        Save Item
      </button>
    </div>
  </div>
</styled.div>
`;

exports[`<ModalAdmin /> should render properly 4`] = `
<styled.div
  className="modal-container"
  modalType="failure"
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.3)",
      "bottom": 0,
      "left": 0,
      "padding": 50,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="card modal"
    id="modalID"
    style={
      Object {
        "display": "block",
        "left": "37%",
        "padding": 0,
        "top": "30%",
        "width": "500px",
      }
    }
  >
    <div
      className="card-toolbar"
      style={
        Object {
          "backgroundColor": "#DA3832",
          "color": "#fff",
          "marginBottom": 0,
          "padding": "10px",
        }
      }
    >
      <span
        className="card-title"
      >
        <h3>
          <i
            className="icon_ban"
          />
           Edit cost of part #123
        </h3>
      </span>
      <i
        className="icon_close"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="card-content"
      style={
        Object {
          "paddingRight": "20px",
        }
      }
    >
      content
    </div>
    <div
      className="card-actions right"
      style={
        Object {
          "margin": 0,
          "padding": "10px",
        }
      }
    >
      <button
        className="button"
        id="close"
        key="0"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button primary"
        id="save"
        key="1"
        onClick={[MockFunction]}
      >
        Save Item
      </button>
    </div>
  </div>
</styled.div>
`;

exports[`<ModalAdmin /> should render properly 5`] = `
<styled.div
  className="modal-container"
  modalType="failure"
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.3)",
      "bottom": 0,
      "left": 0,
      "padding": 50,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="card modal"
    id="modalID"
    style={
      Object {
        "display": "block",
        "left": "37%",
        "padding": 0,
        "top": "0%",
        "width": "500px",
      }
    }
  >
    <div
      className="card-toolbar"
      style={
        Object {
          "backgroundColor": "#DA3832",
          "color": "#fff",
          "marginBottom": 0,
          "padding": "10px",
        }
      }
    >
      <span
        className="card-title"
      >
        <h3>
          <i
            className="icon_ban"
          />
           Edit cost of part #123
        </h3>
      </span>
      <i
        className="icon_close"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="card-content"
      style={
        Object {
          "paddingRight": "20px",
        }
      }
    >
      content
    </div>
    <div
      className="card-actions right"
      style={
        Object {
          "margin": 0,
          "padding": "10px",
        }
      }
    >
      <button
        className="button"
        id="close"
        key="0"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button primary"
        id="save"
        key="1"
        onClick={[MockFunction]}
      >
        Save Item
      </button>
    </div>
  </div>
</styled.div>
`;

exports[`<ModalAdmin /> should render properly 6`] = `
<styled.div
  className="modal-container"
  modalType="failure"
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.3)",
      "bottom": 0,
      "left": 0,
      "padding": 50,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="card modal"
    id="modalID"
    style={
      Object {
        "display": "block",
        "left": "37%",
        "padding": 0,
        "top": "0%",
        "width": "500px",
      }
    }
  >
    <div
      className="card-toolbar"
      style={
        Object {
          "backgroundColor": "#DA3832",
          "color": "#fff",
          "marginBottom": 0,
          "padding": "10px",
        }
      }
    >
      <span
        className="card-title"
      >
        <h3>
          <i
            className="icon_ban"
          />
           Edit cost of part #123
        </h3>
      </span>
      <i
        className="icon_close"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="card-content"
      style={
        Object {
          "paddingRight": "20px",
        }
      }
    >
      content
    </div>
    <div
      className="card-actions right"
      style={
        Object {
          "margin": 0,
          "padding": "10px",
        }
      }
    >
      <button
        className="button"
        disabled="disabled"
        id="close"
        key="0"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        className="button primary"
        id="save"
        key="1"
        onClick={[MockFunction]}
      >
        Save Item
      </button>
    </div>
  </div>
</styled.div>
`;

exports[`<ModalAdmin /> should render properly 7`] = `
.c0 .card .card-toolbar i {
  color: #fff !important;
  font-size: 20px;
  margin-top: 10px;
}

<div
  className="c0"
/>
`;
